static/app/views/insights/common/queries/useTopNDiscoverMultiSeries.ts

```diff
@@ -0,0 +1,225 @@
+  YAxisFields extends SpanMetricsProperty[],
+  Fields extends EapSeriesProperties,
+>(
+  options: UseMetricsSeriesOptions<YAxisFields, Fields>,
+  referrer: string
+) => {
+  return useTopNDiscoverMultiSeries<YAxisFields, Fields>(
+    options,
+    DiscoverDatasets.SPANS_EAP_RPC,
+    referrer
+  );
+};
+
+export const useTopNMetricsMultiSeries = <
+  YAxisFields extends MetricsProperty[],
+  Fields extends MetricsProperty[],
+>(
+  options: UseMetricsSeriesOptions<YAxisFields, Fields>,
+  referrer: string,
+  pageFilters?: PageFilters
+) => {
+  const useEap = useInsightsEap();
+  return useTopNDiscoverMultiSeries<YAxisFields, Fields>(
+    options,
+    useEap ? DiscoverDatasets.SPANS_EAP_RPC : DiscoverDatasets.METRICS,
+    referrer,
+    pageFilters
+  );
+};
+
+/**
+ * TODO: Remove string type, added to fix types for 'count()'
+ */
+export const useTopNSpanIndexedMultiSeries = <
+  YAxisFields extends SpanIndexedField[] | SpanFunctions[] | string[],
+  Fields extends SpanIndexedField[] | SpanFunctions[] | string[],
+>(
+  options: UseMetricsSeriesOptions<YAxisFields, Fields>,
+  referrer: string,
+  dataset?: DiscoverDatasets
+) => {
+  const useEap = useInsightsEap();
+  return useTopNDiscoverMultiSeries<YAxisFields, Fields>(
+    options,
+    useEap ? DiscoverDatasets.SPANS_EAP_RPC : (dataset ?? DiscoverDatasets.SPANS_INDEXED),
+    referrer
+  );
+};
+
+const useTopNDiscoverMultiSeries = <
+  YAxisFields extends string[],
+  Fields extends string[],
+>(
+  options: UseMetricsSeriesOptions<YAxisFields, Fields>,
+  dataset: DiscoverDatasets,
+  referrer: string,
+  pageFilters?: PageFilters
+) => {
+  const {
+    search = undefined,
+    yAxis,
+    fields = [],
+    topN,
+    interval = undefined,
+    samplingMode = DEFAULT_SAMPLING_MODE,
+  } = options;
+
+  const defaultPageFilters = usePageFilters();
+  const location = useLocation();
+  const organization = useOrganization();
+
+  // TODO: remove this check with eap
+  const shouldSetSamplingMode = dataset === DiscoverDatasets.SPANS_EAP_RPC;
+
+  const eventView = getSeriesEventView(
+    search,
+    [...fields, ...yAxis],
+    pageFilters || defaultPageFilters.selection,
+    yAxis,
+    topN,
+    dataset
+  );
+
+  if (interval) {
+    eventView.interval = interval;
+  }
+
+  const result = useGenericDiscoverQuery<MultiSeriesEventsStats, DiscoverQueryProps>({
+    route: 'events-stats',
+    eventView,
+    location,
+    orgSlug: organization.slug,
+    getRequestPayload: () => ({
+      ...eventView.getEventsAPIPayload(location),
+      yAxis: eventView.yAxis,
+      topEvents: eventView.topEvents,
+      excludeOther: 0,
+      partial: 1,
+      orderby: eventView.sorts?.[0] ? encodeSort(eventView.sorts?.[0]) : undefined,
+      interval: eventView.interval,
+      transformAliasToInputFormat: options.transformAliasToInputFormat ? '1' : '0',
+      sampling:
+        samplingMode === 'NONE' || !shouldSetSamplingMode ? undefined : samplingMode,
```

Comment:
Do we set the sampling mode to `NONE` anywhere? It's not a valid sampling mode on the EAP platform so I think it'd be better to keep it as 1:1 as possible

If this is a metrics query and the sampling mode is not defined, then `shouldSetSamplingMode` would set this to undefined anyways so we should be good

Reply:
> Do we set the sampling mode to NONE anywhere? 
Not yet, but if we did i think it feel akward to set `samplingMode: undefined` instead of `samplingMode: 'None'`.

If we did `Normal` by default on the backend, we'll likely have to add a `None` option, and at that point we'll have a one to one mapping for sampling mode

Reply:
the `NONE` option is technically `HIGH_ACCURACY` mode. In the future the platform wants undefined queries to automatically be normal mode and only use the unsampled mode as an explicit request

Reply:
Ahh, i didn't know that, i'll update this in another PR